Users/binah/access-control/access-control.cs

```diff
@@ -0,0 +1,320 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Sansar.Script;
+using Sansar.Simulation;
+using System.Diagnostics;
+
+// register agents on entry
+
+public class AccessControl : SceneObjectScript
+{
+    #region EditorProperties
+    [Tooltip("Comma separated list of user handles for users that are allowed to ban other users.\nThe scene owner is always an admin and does not need to be on this list.")]
+    [DisplayName("Admin Handles")]
+    public string AdminHandles;
+
+    [DefaultValue(true)]
+    [Tooltip("Allow anyone to enter without an access check")]
+    [DisplayName("DoorsOpen")]
+    public bool DoorsOpen;
+
+    [DefaultValue("/showlog")]
+    [Tooltip("Shows visitor log, door open state and banlist")]
+    [DisplayName("Chat Command")]
+    public readonly string VisitorListCommand = "/showlog";
+
+    [DefaultValue("/toggleDoor")]
+    [Tooltip("Toggle door open state")]
+    [DisplayName("Chat Command")]
+    public readonly string ToggleDoorCommand = "/toggleDoor";
+
+    [DefaultValue("/ban")]
+    [Tooltip("Command to ban a user")]
+    [DisplayName("Chat Command")]
+    public readonly string BanCommand = "/ban";
+
+    [DefaultValue("/unban")]
+    [Tooltip("Command to remove a user from banlist")]
+    [DisplayName("Chat Command")]
+    public readonly string UnBanCommand = "/unban";
+
+    [DisplayName("Banned Destination")]
+    [Tooltip("The destination to send banned users.")]
+    [DefaultValue("https://atlas.sansar.com/experiences/katylina/you-have-been-expelled")]
+    public string BannedDestination;
+
+    [DefaultValue(true)]
+    public bool DebugLogging;
+    #endregion
+
+    private RigidBodyComponent _rb;
+
+    private List<string> Admins = new List<string>(); // handle
+    private List<string> Banned = new List<string>(); 
```

Comment:
A `HashSet` might be more appropriate here, since `Contains` is instant, versus looping through entire list in `List.Contains`.

Reply:
Thank you for the help with the language here 🙏